CSharp/BatchExplorer/ViewModels/MainViewModel.cs

```diff
@@ -739,21 +739,84 @@ public CommandBase DeleteSelectedJobs
         {
             get
             {
-                return new CommandBase(
-                    (o) =>
-                    {
-                        foreach (var job in this.jobs)
+                return new CommandBase( 
+                    async (commandArg) =>
+                    {                        
+                        if (this.jobs.Any(a => a.IsChecked))
                         {
-                            if (job.IsChecked || (o != null && o.ToString() == "All"))
+                            IEnumerable<string> deletedIds = this.jobs.Where(a => a.IsChecked).Select(s=>s.Id);
+                            var result =
+                            MessageBox.Show(
+                                deletedIds.Count() > 1 ? 
+                                $"Are you sure you want to delete {deletedIds.Count()} selected jobs?" :

```

Comment:
C# 6 feature - the .sln file is not currently configured to load VS2015 so this will not compile if someone double clicks the .sln (though it will work if they load the .sln into VS2015 explicitly)